warp10/src/main/java/io/warp10/standalone/StandaloneAcceleratedStoreClient.java

```diff
@@ -0,0 +1,298 @@
+      long activityWindow = Long.parseLong(WarpConfig.getProperty(Configuration.INGRESS_ACTIVITY_WINDOW, "-1"));
+      if (activityWindow > 0) {
+        request.setActiveAfter(then / Constants.TIME_UNITS_PER_MS - activityWindow - 1L);
+      }
+    }
+    
+    try {
+      final AtomicLong datapoints = new AtomicLong();
+      
+      MetadataIterator iter = dir.iterator(request);
+      
+      int BATCH_SIZE = Integer.parseInt(WarpConfig.getProperty(Configuration.ACCELERATOR_PRELOAD_BATCHSIZE, "1000"));
+      List<Metadata> batch = new ArrayList<Metadata>(BATCH_SIZE);
+            
+      int nthreads = Integer.parseInt(WarpConfig.getProperty(Configuration.ACCELERATOR_PRELOAD_POOLSIZE, "8"));
+      
+      ThreadPoolExecutor exec = new ThreadPoolExecutor(nthreads, nthreads, 60L, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(nthreads));
+      final AtomicReference<Throwable> error = new AtomicReference<Throwable>();
+      
+      while(iter.hasNext()) {
+        batch.add(iter.next());
+        
+        if (null != error.get()) {
+          throw new RuntimeException("Error populating the accelerator", error.get());
+        }
+        
+        if (BATCH_SIZE == batch.size()) {
+          
+          final List<Metadata> fbatch = batch;
+          
+          Runnable runnable = new Runnable() {            
+            @Override
+            public void run() {
+              try {
+                GTSDecoderIterator decoders = persistent.fetch(null, fbatch, now, then, count, 0, 1.0D, false, 0, 0);
+                
+                while(decoders.hasNext()) {
+                  GTSDecoder decoder = decoders.next();
+                  decoder.next();
+                  GTSEncoder encoder = decoder.getEncoder(true);
+                  cache.store(encoder);
+                  datapoints.addAndGet(decoder.getCount());
+                }                              
+              } catch (Exception e) {
+                error.set(e);
+                throw new RuntimeException(e);
+              }
+            }
+          };
+          
+          boolean submitted = false;
+          while(!submitted) {
+            try {
+              exec.execute(runnable);
+              submitted = true;
+            } catch (RejectedExecutionException re) {
+              LockSupport.parkNanos(100000000L);
+            }
+          }
+          
+          batch = new ArrayList<Metadata>(BATCH_SIZE);
+        }
+      }
+
+      if (!batch.isEmpty()) {          
+        GTSDecoderIterator decoders = this.persistent.fetch(null, batch, now, then, count, 0, 0.0, false, 0, 0);
+        
+        while(decoders.hasNext()) {
+          GTSDecoder decoder = decoders.next();
+          decoder.next();
+          GTSEncoder encoder = decoder.getEncoder(true);
+          this.cache.store(encoder);
+          datapoints.addAndGet(decoder.getCount());
+        }
+      }
+
+      exec.shutdown();
```

Comment:
You have to wait for the tasks to complete after that. If you want to log the waiting, you can do:
```
while (!exec.awaitTermination(30, TimeUnit.SECONDS)) {
  LOG.info("Awaiting completion of accelerator preloading.");
}
```